prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,328 @@
+                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                    ?: throw Throwable("Error in validation of jwt token")
+                // get the user name and org
+                return jwt.claims["sub"].toString();
+            }
+            catch (ex: Throwable) {
+                context.logger.log(Level.WARNING, "Error in verification of token", ex)
+                return null
+            }
+        }
+        return null;
+    }
+    
+
+    /**
+     * TODO: Fixme!
+     */
+    private fun getSchedules(): List<String>{
+        return WorkflowEngine.databaseAccess.fetchEmailSchedules();
+    }
+
+    /**
+     * Converts an organization name to an OKTA group name
+     */
+    private fun convertOrgToGroup( org: String ): String {
+        return "DH"+org.replace( "-","_" )
+    }
+
+    /**
+     * Retrieve a list of emails within an organization
+     * 
+     * @params org The organization to fetch emails for
+     * @returns List of emails to send to
+     */
+    private fun getEmails( org: String, context: ExecutionContext ): List<String> {
+
+        var ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
+
+        if( ssws == null ){
+            context.logger.warning("Can't find SSWS_OKTA secret")
+            return listOf();
+        }
+        
+        var grp = convertOrgToGroup( org );
+
+        // get the OKTA Group Id
+        var response1 = httpGet( url="https://hhs-prime-admin.okta.com/api/v1/groups?q=${grp}", 
+                               headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+        var grpId = ((response1.jsonArray).get( 0 ) as JSONObject).getString("id");
+
+        // get the users within that OKTA group
+        var response = httpGet( url="https://hhs-prime-admin.okta.com/api/v1/groups/${grpId}/users", 
+                               headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+    
+       var emails :MutableList<String> = mutableListOf()
+
+        for( user in response.jsonArray ){
+            emails.add( (user as JSONObject).getJSONObject("profile").getString("email") );
+        } 
+
+        return emails;
+    }
+
+     /**
+     * Get the reports that have been generated since a given date
+     * 
+     * @param org
+     * 
+     */
+    private fun getReportsSinceLast( org: String, last: Date ): List<ReportFile> {
+
+      val reportFiles = workflowEngine
+                        .fetchDownloadableReportFiles( 
+                                last.toInstant().atOffset(ZoneOffset.UTC), 
+                                org )
+
+        return reportFiles;
+    }
+
+     /**
+     * TODO: Fixme!
+     */
+    @Suppress( "UNRESOLVED_REFERENCE")
+    private fun dispatchToSendGrid( 
+        template: String,
+        emails: List<String>,
```

Comment:
Nitpick: this method only ever uses the `emails` to iterate over it; consider asking for it as an `Iterable<String>` instead of as a list.
That makes your design intentions clear: I am asking for something and will only iterate over it. Any changes to behavior in the method's implementation will now show up as a clear deviation to that initial design because it will require that you change the parameter type.
Think of this as both broadening your inputs and tightening your own operations:
- Broadening: more things can be used to call this method (more things are `Iterable<String>` than just `List<String>` so you can now call this method with _any_ of those
- tightening: this method intends to only iterate over emails, by asking for it as an `Iterable<String>`, you ensure that that is all you _can_ do and you give a guarantee to your caller that you won't modify this thing in any way.

Reply:
Its not always the case that it will only be iterated over...this is a first cut at the function for later modifications

Reply:
Are we expecting in the future to do more than iterate over it? Based on the logic and name, it doesn't appear that way.
I guess generally speaking: yes, I agree (with caveats), but here, I would change it.